server/src/main/java/org/apache/druid/client/ImmutableDruidDataSource.java

```diff
@@ -120,8 +120,32 @@ public String toString()
            + "'}";
   }
 
+  /**
+   *
+   * Look at https://github.com/apache/incubator-druid/issues/7858 for explanation of why equals method is not supported.
+   *
+   */
   @Override
   public boolean equals(Object o)
+  {
+    throw new UnsupportedOperationException("equals:ImmutableDruidDataSource shouldn't be used as the key in containers");
```

Comment:
It looks like `equals:ImmutableDruidDataSource` is not the Java convention. At least I'm not familiar with it. Would you tell me where this convention came from?

Reply:
My bad. It isn't any convention. Wrote equals: prefix for debugging and missed removing that.

Reply:
Done.